scripts/postgres/schema-v1-postgresql.sql

```diff
@@ -0,0 +1,94 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one
+-- or more contributor license agreements.  See the NOTICE file--
+--  distributed with this work for additional information
+-- regarding copyright ownership.  The ASF licenses this file
+-- to you under the Apache License, Version 2.0 (the
+-- "License"). You may not use this file except in compliance
+-- with the License.  You may obtain a copy of the License at
+--
+--  http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing,
+-- software distributed under the License is distributed on an
+-- "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+-- KIND, either express or implied.  See the License for the
+-- specific language governing permissions and limitations
+-- under the License.
+--
+
+-- Note: Database and schema creation is not included in this script. Please create the database and
+-- schema before running this script. for example in psql:
+-- CREATE DATABASE polaris_db;
+-- \c polaris_db
+-- CREATE SCHEMA polaris_schema;
+-- set search_path to polaris_schema;
+
+CREATE TABLE IF NOT EXISTS entities (
+    catalog_id BIGINT NOT NULL,
+    id BIGINT NOT NULL,
+    parent_id BIGINT NOT NULL,
+    name TEXT NOT NULL,
+    entity_version INT NOT NULL,
+    type_code INT NOT NULL,
+    sub_type_code INT NOT NULL,
+    create_timestamp BIGINT NOT NULL,
+    drop_timestamp BIGINT NOT NULL,
+    purge_timestamp BIGINT NOT NULL,
+    to_purge_timestamp BIGINT NOT NULL,
+    last_update_timestamp BIGINT NOT NULL,
+    properties JSONB not null default '{}'::JSONB,
+    internal_properties JSONB not null default '{}'::JSONB,
+    grant_records_version INT NOT NULL,
+    PRIMARY KEY (id),
+    CONSTRAINT constraint_name UNIQUE (catalog_id, parent_id, type_code, name)
+);
+
+-- TODO: create indexes based on all query pattern.
+CREATE INDEX IF NOT EXISTS idx_entities ON entities (catalog_id, id);
```

Comment:
Might as well make this the primary key.... I guess all access paths always provide a `catalog_id` (even if it is `0`).

Including `catalog_id` in the PK adds extra protection against (randomly generated) ID collisions.

Reply:
Presently we have `id` as PK, we guarantee, `id` unique across so if we change PK `(catalog_id, id)` so (0, id) and (1, id) is acceptable which shouldn't be the case, hence moved this to index, based on discussion [here](https://github.com/apache/polaris/pull/1276/files#r2021737361)

Reply:
Agreed, entity ID is the PK as we have a uniqueness constraint on entity ID

Reply:
> [...] since we guarantee, id unique across

My understanding is that it is not the job of Persistence to validate ID uniqueness. Still, does a PR on ID really help in preventing ID collisions from the perspective of the `BasePersistence` interface? Inputs into that interfaces may be rather arbitrary.

That said, I do not really object to having this index, if you prefer to keep it.

Reply:
> Still, does a PR on ID really help in preventing ID collisions from the perspective of the BasePersistence interface? Inputs into that interfaces may be rather arbitrary.

Thats true BasePersistence independently generates the ID, and the expectation is that it should not conflict, we just gate even if it does by adding the PK restriction